sched/pthread/pthread_barrierwait.c

```diff
@@ -82,67 +82,35 @@
 
 int pthread_barrier_wait(FAR pthread_barrier_t *barrier)
```

Comment:
let's create a new patch moving pthread_barrierwait.c to libc/pthread, since the new implementation doesn't call the kernel internal api.